test/test_ion_text.cpp

```diff
@@ -184,14 +193,15 @@ TEST(IonTextSymbol, ReaderReadsAnnotationSymbolZero) {
     ION_STRING annotation_strs[1];
     SIZE num_annotations;
     ION_SYMBOL symbol;
+    char *tmp = NULL;
```

Comment:
The buffer allocated by `ion_string_strdup` needs to be freed. The `tmp` variable will receive the address to the buffer, so it can be freed once the comparison is done, and re-used for each instance.